application/src/main/java/org/opentripplanner/apis/gtfs/datafetchers/LegImpl.java

```diff
@@ -275,8 +275,17 @@ private Leg getSource(DataFetchingEnvironment environment) {
     return environment.getSource();
   }
 
+  @Override
+  public DataFetcher<Iterable<Leg>> previousLegs() {
+    return nextOrPreviousLegs(true);
+  }
+
   @Override
   public DataFetcher<Iterable<Leg>> nextLegs() {
+    return nextOrPreviousLegs(false);
+  }
+
+  private DataFetcher<Iterable<Leg>> nextOrPreviousLegs(boolean includeDepartBefore) {
```

Comment:
This `includeDepartBefore` is a slightly difficult to understand boolean name (I know the name is copied over from the `getAlternativeLegs` method) but I think you could give it some more understandable name like `previousLegsOnly`. In general, we try to avoid boolean method variables that change the behavior. Instead, we prefer to split the methods into two different methods where the name reflects what it does. However, in this case you are just repeating the design decision made in the `getAlternativeLegs` method so it might be ok. We can discuss this in a developer meeting.

Reply:
We discussed this in the developer meeting and came to the conclusion that you should split this private method and also getAlternativeLegs into two methods. One for previous legs and one for next legs.

Reply:
The `previousLegs` and `nextLegs` are doing the same thing, with the only difference in direction. I don't want to duplicate codes so I will switch to an enum instead.

Reply:
Enum isn't really any better than boolean. You can put some of the code that computes the other arguments into methods that can be used by both the nextLegs and previousLegs methods so you can avoid some duplication that way.

Reply:
I don't think I can do anything better than converting the boolean to an enum, as the result will be much more complicated.

The single flag is used for multiple purposes:

- to sort the departures within a pattern
- to filter the results to be before / after the requested time
- to sort the leg results after combining them from different patterns

If I don't use such a flag, I will need two more parameters for `generateLegs`, one to sort the results, one to compare the stop times in the priority queue. I will also need another parameter for `getAlternativeLegs` to sort the legs across different trip patterns as well, so it will be 3 different interrelated new arguments passed down the call stack instead of a single flag, if these arguments become out of sync strange behavior will result.

`nextLegs` and `previousLegs` are basically the same thing. They are not different behavior. The same argument can be said of journey planning by departure time or arrival time which should be the same logic, yet I am seeing bugs which happens on arrival only but not departure (#6102).